http/oidc/pom.xml

```diff
@@ -0,0 +1,154 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+  ~ JBoss, Home of Professional Open Source.
+  ~ Copyright 2020 Red Hat, Inc., and individual contributors
+  ~ as indicated by the @author tags.
+  ~
+  ~ Licensed under the Apache License, Version 2.0 (the "License");
+  ~ you may not use this file except in compliance with the License.
+  ~ You may obtain a copy of the License at
+  ~
+  ~     http://www.apache.org/licenses/LICENSE-2.0
+  ~
+  ~ Unless required by applicable law or agreed to in writing, software
+  ~ distributed under the License is distributed on an "AS IS" BASIS,
+  ~ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  ~ See the License for the specific language governing permissions and
+  ~ limitations under the License.
+  -->
+
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+
+    <parent>
+        <groupId>org.wildfly.security</groupId>
+        <artifactId>wildfly-elytron-parent</artifactId>
+        <version>1.17.0.CR1-SNAPSHOT</version>
+        <relativePath>../../pom.xml</relativePath>
+    </parent>
+
+    <modelVersion>4.0.0</modelVersion>
+
+    <artifactId>wildfly-elytron-http-oidc</artifactId>
```

Comment:
Seems to me majority of clases of this MR are OIDC specification implementation (no dependancy on Elytron) and just small amount of classes, e.g. ~ 7 dependant on Elytron. Is it possible to raise OIDC implementation to its own module/library? And such component would be sharable between other projects.

Honestly when looking into Keyckloak list of adapters[1], I would expect Keyckloak will provide such library. Because I assume every adapter share this OIDC specification implementation.

What do you think? Have you discussed that before?

[1] https://www.keycloak.org/docs/latest/securing_apps/#java

Reply:
@mchoma One of the main motivations for this enhancement is to stop the need for Keycloak to be providing client adapters for WildFly and JBoss EAP - additionally the ones provided here are usable with more prodiders than just Keycloak.

Reply:
I understand that and agree. New native subsystem and integration with Elytron makes sense to me. 

I am talking about code which seems is pure implementation of  OIDC specification, e.g. classes like this [1]. Is it worth to pull it on one place and share it among different projects? And not to have copies on different projects?

[1] https://github.com/wildfly-security/wildfly-elytron/pull/1569/files#diff-7db38f7db99790ce5017a9d995ea6933bcb3b3f3c3f85a1311d386b047bec317

Reply:
@mchoma We can look at moving specific classes out to their own module if projects like Keycloak still need to make use of these kinds of classes even after they stop providing the client adapters. We'd need to check each class carefully to see if it makes sense or not. A lot of Keycloak's current classes are specific to Keycloak whereas our implementation is meant to be used with any OpenID provider so in a lot of cases, it's not a straightforward copy/paste of an existing class from Keycloak.

Reply:
As we are not adding as public API I think it is safe to add to Elytron and remove later if needed.

There were some discussions in the past about moving ACME support to SmallRye, maybe overall there could be a smallrye-security project with a few independent modules which can be shared - OpenID connect could be proposed there.  But I think these kind of thoughts will be better as a follow up later.